generator/template.go

```diff
@@ -195,6 +195,91 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+/**
+ * Flattens a deeply nested object
+ *
+ * @param  {Object} deeplyNestedObject
+ * @param  {String} path
+ * @return {Record<string, unknown>}
+ */
+function flattenObject<T extends Record<string, unknown>>(
+  deeplyNestedObject: T,
+  path: string = ""
+): T {
+  return Object.keys(deeplyNestedObject).reduce(
+    (acc: T, key: string): T => {
+      const value = deeplyNestedObject[key];
+
+      const newPath = Array.isArray(deeplyNestedObject)
+        ? path + "." + key
+        : [path, key].filter(Boolean).join(".");
+      const isObject = [
```

Comment:
Sure I will check it out, thanks for your feedback! 🙇🏼

Reply:
@lyonlai I did some more research and found a simpler one line way of checking if something is a plain object.

```js
const isPlainObject = Object.prototype.toString.call(value) === "[object Object]" ? true : false;
```

The following are results of calling ` Object.prototype.toString.call(value)` on various types of values.
```
{"foo": bar, "baz": {1, 2}} => [object Object]
{"foo": 1} => [object Number]
{"foo": undefined} => [object Undefined]
{"foo: [1, 2]} => [object Array]
{"foo: null} => [object Null]
```
etc...

Let me know what you think about this 🙏🏼

Reply:
Next time could you please also show the source of where you get the idea from? also good for the reviewer to verify the idea. 

the message is confusing, calling `Object.prototype.toString` won't get the result you posted. need to call the function like `Object.prototype.toString.call(1)` then will get the result.

Though it's simple, however 

```
class A {}
var a = new A()
Object.prototype.toString.call(a)
```

will get the result: `"[object Object]"`.  which makes this simple implementation not sufficient. 

Please make a copy of the `isPlainObject.js` as stated above. it covers quite a few edge cases.

Reply:
Thank you for your quick feedback. My apologies for not noticing the above edge case. 
As suggested I will use the transpiled version `isPlainObject.js` as the function itself imports other ramda functions within it 🙇🏼

Reply:
yup, it imports other `ramda` functions for sure. but there aren't complex logic to replicate though. let's see

Reply:
This is the final form of the `isPlainObj.js` (latest version) from `ramda-adjunct` because `ramda` library does not contain a function called `isPlainObject.js`. It imports a lot of helpers from `ramda` and those helpers import other helper functions as well. Is it possible to include the `ramda.min.js` as a separate generated file?

Link for source: https://github.com/char0n/ramda-adjunct/blob/master/src/isPlainObj.js


```js
import {
  pipe,
  type,
  identical,
  both,
  equals,
  toString,
  pathSatisfies,
  curryN,
  anyPass,
  complement,
  is,
  either,
  stubFalse
} from "ramda";

const isOfTypeObject = val => typeof val === "object";

/**
 * Checks if input value is `null`.
 *
 * @func isNull
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.1.0|v0.1.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isNotNull|isNotNull}
 * @example
 *
 * RA.isNull(1); //=> false
 * RA.isNull(undefined); //=> false
 * RA.isNull(null); //=> true
 */
const isNull = equals(null);

/**
 * Checks if input value is complement of `null`.
 *
 * @func isNotNull
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.1.0|v0.1.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isNull|isNull}
 * @example
 *
 * RA.isNotNull(1); //=> true
 * RA.isNotNull(undefined); //=> true
 * RA.isNotNull(null); //=> false
 */
const isNotNull = complement(isNull);

/* eslint-disable max-len */
/**
 * Checks if value is object-like. A value is object-like if it's not null and has a typeof result of "object".
 *
 * @func isObjLike
 * @aliases isObjectLike
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.5.0|v0.5.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isNotObjLike|isNotObjLike}, {@link RA.isObj|isObj}, {@link RA.isPlainObj|isPlainObj}
 * @example
 *
 * RA.isObjLike({}); //=> true
 * RA.isObjLike([]); //=> true
 * RA.isObjLike(() => {}); //=> false
 * RA.isObjLike(null); //=> false
 * RA.isObjLike(undefined); //=> false
 */
/* eslint-enable max-len */
const isObjLike = curryN(1, both(isNotNull, isOfTypeObject));

let GeneratorFunction = null;
let legacyCheck = null;
try {
  GeneratorFunction = new Function("return function* () {}")().constructor; // eslint-disable-line no-new-func
  legacyCheck = is(GeneratorFunction);
} catch (e) {
  legacyCheck = stubFalse;
}

/**
 * Checks if input value is `Generator Function`.
 *
 * @func isGeneratorFunction
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.5.0|v0.5.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isFunction|isFunction}, {@link RA.isAsyncFunction|isAsyncFunction}, {@link RA.isNotGeneratorFunction|isNotGeneratorFunction}
 * @example
 *
 * RA.isGeneratorFunction(function* test() { }); //=> true
 * RA.isGeneratorFunction(null); //=> false
 * RA.isGeneratorFunction(function test() { }); //=> false
 * RA.isGeneratorFunction(() => {}); //=> false
 */
const isGeneratorFunction = curryN(
  1,
  either(
    pipe(
      type,
      identical("GeneratorFunction")
    ),
    legacyCheck
  )
);

/**
 * Checks if input value is `Async Function`.
 *
 * @func isAsyncFunction
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.5.0|v0.5.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isFunction|isFunction}, {@link RA.isNotAsyncFunction|isNotAsyncFunction}, {@link RA.isGeneratorFunction|isGeneratorFunction}
 * @example
 *
 * RA.isAsyncFunction(async function test() { }); //=> true
 * RA.isAsyncFunction(null); //=> false
 * RA.isAsyncFunction(function test() { }); //=> false
 * RA.isAsyncFunction(() => {}); //=> false
 */
const isAsyncFunction = curryN(
  1,
  pipe(
    type,
    identical("AsyncFunction")
  )
);

/**
 * Checks if input value is `Function`.
 *
 * @func isFunction
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.5.0|v0.5.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isNotFunction|isNotFunction}, {@link RA.isAsyncFunction|isNotAsyncFunction}, {@link RA.isGeneratorFunction|isGeneratorFunction}
 * @example
 *
 * RA.isFunction(function test() { }); //=> true
 * RA.isFunction(function* test() { }); //=> true
 * RA.isFunction(async function test() { }); //=> true
 * RA.isFunction(() => {}); //=> true
 * RA.isFunction(null); //=> false
 * RA.isFunction('abc'); //=> false
 */
const isFunction = anyPass([
  pipe(
    type,
    identical("Function")
  ),
  isGeneratorFunction,
  isAsyncFunction
]);

const isObject = pipe(
  type,
  identical("Object")
);
const isObjectConstructor = pipe(
  toString,
  equals(toString(Object))
);
const hasObjectConstructor = pathSatisfies(
  both(isFunction, isObjectConstructor),
  ["constructor"]
);

/* eslint-disable max-len */
/**
 * Check to see if an object is a plain object (created using `{}`, `new Object()` or `Object.create(null)`).
 *
 * @func isPlainObj
 * @aliases isPlainObject
 * @memberOf RA
 * @since {@link https://char0n.github.io/ramda-adjunct/0.5.0|v0.5.0}
 * @category Type
 * @sig * -> Boolean
 * @param {*} val The value to test
 * @return {boolean}
 * @see {@link RA.isNotPlainObj|isNotPlainObj}, {@link RA.isObjLike|isObjLike}, {@link RA.isObj|isObj}
 * @example
 *
 * class Bar {
 *   constructor() {
 *     this.prop = 'value';
 *   }
 * }
 *
 * RA.isPlainObj(new Bar()); //=> false
 * RA.isPlainObj({ prop: 'value' }); //=> true
 * RA.isPlainObj(['a', 'b', 'c']); //=> false
 * RA.isPlainObj(Object.create(null); //=> true
 * RA.isPlainObj(new Object()); //=> true
 */
/* eslint-enable max-len */
const isPlainObj = curryN(1, val => {
  if (!isObjLike(val) || !isObject(val)) {
    return false;
  }

  const proto = Object.getPrototypeOf(val);

  if (isNull(proto)) {
    return true;
  }

  return hasObjectConstructor(proto);
});

export default isPlainObj;
```

Reply:
it seems complex at the first glance. but essentially the logic is quite simple to copy  and the imported ramda functions are not necessary. 

isObject
```
a => Object.prototype.toString.call({a: 1}).slice(8, -1) === 'Object'
```

isObjLike
```
a => a !== null && isObject
```

hasObjectConstructor
```
// give proto.constructor as input
typeof input === 'function' && input === Object.prototype.constructor
```

Reply:
> it seems complex at the first glance. but essentially the logic is quite simple to copy and the imported ramda functions are not necessary.
> 
> isObject
> 
> ```
> a => Object.prototype.toString.call({a: 1}).slice(8, -1) === 'Object'
> ```
> 
> isObjLike
> 
> ```
> a => a !== null && isObject
> ```
> 
> hasObjectConstructor
> 
> ```
> // give proto.constructor as input
> typeof input === 'function' && input === Object.prototype.constructor
> ```


I see. I will create a seperate `isObject` function based on your feedback code 🙇🏼

Reply:
@lyonlai 
Based on your feedback, I have created a `isPlainObject` function.
```ts
function isPlainObject(val: unknown): boolean {
  const isObject =
    Object.prototype.toString.call(val).slice(8, -1) === "Object";
  const isObjLike = val !== null && isObject;

  if (!isObjLike || !isObject) {
    return false;
  }
  
  const proto = Object.getPrototypeOf(val);

  const hasObjectConstructor =
    typeof proto === "object" &&
    proto.constructor === Object.prototype.constructor;

  console.log(typeof proto === "function");

  return hasObjectConstructor;
}
```
I tried this `typeof input === 'function' && input === Object.prototype.constructor`. However, this returns `false` even if given input is an object.
Hence I modified it to below logic because according to my understanding Object.prototype should not be a function
```js
  const hasObjectConstructor =
    typeof proto === "object" &&
    proto.constructor === Object.prototype.constructor;
```

 Let me know if I have missed anything 🙇🏼